src/components/DropList/DropList.jsx

```diff
@@ -0,0 +1,230 @@
+import Tippy from '@tippyjs/react/headless'
+import { noop } from '../../utilities/other'
+import { GlobalContext } from '../HSDS/Provider'
+import { OPEN_ACTION_ORIGIN, VARIANTS } from './DropList.constants'
+import {
+  flattenListItems,
+  itemToString,
+  isTogglerOfType,
+  useWarnings,
+} from './DropList.utils'
+import {
+  Button,
+  SelectTag,
+  getTogglerPlacementProps,
+} from './DropList.togglers'
+import Animate from '../Animate'
+import Combobox from './DropList.Combobox'
+import Select from './DropList.Select'
+
+function DropListManager({
+  animateOptions = {},
+  autoSetComboboxAt = 0,
+  closeOnClickOutside = true,
+  closeOnSelection = true,
+  customEmptyList = null,
+  'data-cy': dataCy,
+  initialIsOpen = false,
+  initialSelectedItem = null,
+  items = [],
+  onOpenedStateChange = noop,
+  onSelect = noop,
+  renderCustomListItem = null,
+  tippyOptions = {},
+  toggler = {},
+  variant = VARIANTS.SELECT,
+  withMultipleSelection = false,
+}) {
+  const parsedItems = flattenListItems(items)
+  const [isOpen, setOpenedState] = useState(initialIsOpen)
+  const [openOrigin, setOpenOrigin] = useState('')
+  const [selectedItem, setSelectedItem] = useState(initialSelectedItem)
+  const { getCurrentScope } = useContext(GlobalContext) || {}
+  const scope = getCurrentScope ? getCurrentScope() : null
+
+  useWarnings({ toggler, withMultipleSelection })
+
+  function onSelectionChange(selection) {
+    onSelect(selection)
+    setSelectedItem(selection)
+  }
+
+  function toggleOpenedState(isOpen, origin = '') {
+    setOpenedState(isOpen)
+    setOpenOrigin(origin)
+    onOpenedStateChange(isOpen)
+  }
+
+  const animateProps = {
+    duration: 200,
+    easing: 'ease-in-out',
+    sequence: 'fade down',
+    ...animateOptions,
+    // These shouldn't be overriden
+    animateOnMount: true,
+    mountOnEnter: false,
+    unmountOnExit: false,
+  }
+  const tippyProps = {
+    interactive: true,
+    placement: 'bottom-start',
+    ...tippyOptions,
+  }
+  let Toggler
+
+  if (React.isValidElement(toggler)) {
+    const { onClick } = toggler.props
+    const togglerProps = {
+      onClick: e => {
+        onClick && onClick(e)
+
+        /**
+         * On combobox when clicking the button a blur event happens first that closes
+         * the DropList, here we check if the menu was closed due to the input blur and ignore the
+         * click action to toggle the open state, we also always reset the "origin" to resume normal behaviour
+         */
+        if (openOrigin !== OPEN_ACTION_ORIGIN.INPUT_BLUR) {
+          toggleOpenedState(!isOpen)
+        }
+        setOpenOrigin('')
+      },
+      isActive: isOpen,
+    }
+
+    if (toggler.type === SelectTag) {
+      const { text } = toggler.props
+
+      if (text == null) {
+        togglerProps.text = itemToString(selectedItem)
+      }
+    }
+
+    const { placement, offset } = getTogglerPlacementProps(toggler)
+    tippyProps.placement = placement
```

Comment:
In case we create a custom toggler, it might be nice to let that component pass down a placement/offset. I guess we could skip that step if tippyProps/tippyOptions already contain a placement/offset

any thoughts on that?